### PR TITLE
Telemetry stickiness issue

### PR DESCRIPTION
**Pull request by JoelMcKinnon**
Moved the persist() calls into the select event listeners so they don't get called on mounted hook.

Cleaned up clearInputs and renamed as clearDependentFields. Used refs to check on change event select element.

Resolves issue: #2770

### CODE REVIEW

File: src/plugins/condition/components/Condition.vue [hunk 1]
```diff
@@ -338,11 +348,8 @@ export default {
             this.condition.configuration.criteria.splice(index + 1, 0, clonedCriterion);
             this.persist()
         },
-        hasTelemetry(identifier) {
-            // TODO: check parent condition.composition.hasTelemetry
-            return this.currentCriteria && identifier;
-        },
         persist() {
+            console.log('persist called from Condition.vue')
```
Comment (davetsay) [inline src/plugins/condition/components/Condition.vue]: remove the logging

Comment (JoelMcKinnon) [reply src/plugins/condition/components/Condition.vue]: done

**Comment by JoelMcKinnon** [comment]
> Having refs and v-models both mutating the same DOM elements should be used with care and most likely isn't the best solution.
> 
> It looks like the problem is with clearing inputs that are related to other inputs. Seems like basically the order in which you build the condition matters, and if you destroy a choice all further choices should be cleared. I think I the v-models are always good choices for user inputs in general. I would keep that and remove the refs. Then, I would take advantage of watches to cascade down clears of dependent inputs.
> 
> Maybe using the refs could be more performant in some ways but I feel it adds a lot of complexity and management to execute.

I wasn't aware the refs would mutate the dom elements. I thought they just mapped them for reference, like classes or ids. I do see a problem though, in that the refs won't be unique in the case of multiple criteria (like with ids). I can solve this with use of closest() and use a js- prefixed classname instead of the ref. Do you see a problem with that? Would watches still be a preferred approach?

**Comment by JoelMcKinnon** [comment]
@davetsay  I found cleaner way of identifying the select elements. Used classes js-telemetry-select and js-metadata-select on the elements and then checked if the event target element passed contained those classes. Also fixed an issue resulting in no and/or in description.

**Comment by davetsay** [comment]
@JoelMcKinnon ,

Sorry today was crazy. Would probably have been best to just chat regarding this. I didn't take a long look at this but it looked like the refs were being used in order to access and directly mutate those elements, which seems like a bad idea if they have v-models on them. We can chat more about this tomorrow.

**Comment by JoelMcKinnon** [comment]
I still had one persist call in Criterion.vue outside of the change event condition. Also was still referring to a $ref in updateOperations and removed that along with an unnecessary $emit..

**Pull request merged**